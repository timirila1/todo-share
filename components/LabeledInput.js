import React from 'react';
import { StyleSheet, View, Text, TextInput } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { colors } from '../services/consts';

export const LabeledInput = props => {
    const { label } = props;

    return (
        <View>
            <Text style={styles.label}>{label}</Text>
            <TextInput
                style={styles.input}
                {...props}
            />
        </View>

    );
};

const styles = StyleSheet.create({
    input: {
        paddingTop: 10,
        paddingRight: 10,
        paddingBottom: 10,
        borderBottomColor: colors.main,
        borderBottomWidth: 1,
        fontSize: 24,
    },
    label: {
        color: colors.textDark,
    }
});
