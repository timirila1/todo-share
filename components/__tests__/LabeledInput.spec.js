import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { LabeledInput } from '../LabeledInput';

it('Render LabeledInput', async () => {
    const tree = renderer.create(
        <LabeledInput label="test" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
