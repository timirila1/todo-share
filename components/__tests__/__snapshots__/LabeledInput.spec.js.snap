// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LabeledInput 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.6)",
      }
    }
  >
    test
  </Text>
  <TextInput
    allowFontScaling={true}
    label="test"
    style={
      Object {
        "borderBottomColor": "#5e1c73",
        "borderBottomWidth": 1,
        "fontSize": 24,
        "paddingBottom": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
